cmd/gh/main.go

```diff
@@ -45,6 +45,18 @@ func main() {
 	stderr := cmdFactory.IOStreams.ErrOut
 	rootCmd := root.NewCmdRoot(cmdFactory, command.Version, command.BuildDate)
 
+	cfg, err := cmdFactory.Config()
+	if err != nil {
+		fmt.Fprintf(stderr, "failed to read configuration:  %s\n", err)
+		os.Exit(2)
```

Comment:
Are there legitimate cases when configuration could not be read, but we want to allow executing commands? E.g. when the config file has a syntax error due to manual edits?

I can't think of any, but if you do think of some cases, it's always an option to write a warning notice to stderr here instead of failing. Right now, I'm fine with failing, since a broken config file means that authentication info could not be read, and a missing config file does not raise any errors.

Nit: since you have it here, you can now avoid reading configuration in the lower `authCheckEnabled` block.

Reply:
I can't think of any either and would prefer to fail; running with a broken config can put users into a place that is very hard for them and us to debug